### PR TITLE
[dependabot] add dependabot config file

### LINKED ISSUE #14956: [python] setup dependabot to automatically update python packages
#14941 pinned python package versions. To address [this comment](https://github.com/lowRISC/opentitan/pull/14941#pullrequestreview-1109045714), we should aim for automatic package version updates to prevent painful upgrade processes in the future.

### PR DESCRIPTION
**Pull request by timothytrippel**
This adds a dependabot config file to trigger automatic updates to Python packages in response to #14941. This fixes #14956.

Signed-off-by: Timothy Trippel <ttrippel@google.com>

### CODE REVIEW

**Comment by timothytrippel** [comment]
As noted here, this will require updates to the repo configs to:
1. activate this
2. maybe assign a reviewer team for PRs generated by this?

@mundaym we can discuss in person next week.

**Comment by alphan** [comment]
@timothytrippel I merged the PR that pins the package versions, you can rebase this one.

File: dependabot.yml
Comment (moidx) [inline dependabot.yml:19]: We can probably work with @mundaym in defining a rotation to review these type of changes. I am not sure if dependabot supports GitHub teams? In the meantime, I'd also add someone from lowRISC here and below.

File: dependabot.yml
Comment (moidx) [inline dependabot.yml:21]: nit: Add a comment describing why we are skipping these repositories. Thanks!

**Review by moidx — APPROVED** [review]

File: dependabot.yml
Comment (moidx) [reply dependabot.yml:19]: @timothytrippel  oops, I missed your previous comment. Feel free to resolve this one.

**Review by alphan — APPROVED** [review]

**Comment by alphan** [comment]
@timothytrippel please let us know if we must wait for the repo settings to be updated before merging this. Thanks!

**Comment by timothytrippel** [comment]
> @timothytrippel please let us know if we must wait for the repo settings to be updated before merging this. Thanks!

We should wait until updating the repo settings b/c I will change the "reviewers" field to something other than myself hehe.

File: dependabot.yml
Comment (timothytrippel) [reply dependabot.yml:21]: fixed.

File: dependabot.yml
Comment (milesdai) [inline dependabot.yml:16]: Can we configure dependabot to add a "Signed off by" line to the commit since one of our lint steps in CI checks for this on each commit?

**Review by milesdai — APPROVED** [review]

File: dependabot.yml
Comment (timothytrippel) [reply dependabot.yml:16]: It is unclear if dependabot will add the signed off line or not. The [documentation](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#commit-message) claims: "_Dependabot attempts to detect your commit message preferences_", but they also give you the option set a prefix. From what I can find, it looks like this is an outstanding [feature request](https://github.com/dependabot/dependabot-core/issues/1460), and [we will have to](https://github.com/dependabot/dependabot-core/issues/1460#issuecomment-545349408) waive the signoff requirement for dependabot PRs.

Comment (timothytrippel) [reply dependabot.yml:16]: However, I also see a [screenshot](https://github.com/dependabot/dependabot-core/issues/1683) from a dependabot commit that indicates we might OK. So maybe let's wait and see what a dependabot commit looks like and go from there?

**Comment by timothytrippel** [comment]
Reviewer team has been created by @mundaym and the dependabot config file has been updated to use this. Ready to merge when this passes CI.

**Pull request merged**
